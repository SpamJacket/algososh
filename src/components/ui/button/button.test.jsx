import renderer from "react-test-renderer";
import { Button } from "./button";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Рендер кнопки", () => {
  it("Кнопка c текстом рендерится без ошибок", () => {
    const tree = renderer.create(<Button text="Текст для теста" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка без текста рендерится без ошибок", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Заблокированная кнопка рендерится без ошибок", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка во время загрузки рендерится без ошибок", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Обработка колбека кнопки", () => {
  it("При клике на кнопку, колбек вызывается корректно", () => {
    window.alert = jest.fn();

    render(<Button onClick={() => alert("Алгоритм запустился!")} />);

    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(window.alert).toHaveBeenCalledWith("Алгоритм запустился!");
  });
});
