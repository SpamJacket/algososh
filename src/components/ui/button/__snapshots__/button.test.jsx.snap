// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендер кнопки Заблокированная кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Рендер кнопки Кнопка c текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст для теста
  </p>
</button>
`;

exports[`Рендер кнопки Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Рендер кнопки Кнопка во время загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary undefined undefined undefined "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    src="test-file-stub"
  />
</button>
`;
